src/d2_docker/config/dhis2-core-entrypoint.sh

```diff
@@ -49,6 +52,18 @@ if [ "$(id -u)" = "0" ]; then
         rm -v $WARFILE # just to save space
     fi
 
+    if [ -f $GLOWROOT_ZIP ] && [ ! -d $GLOWROOT_DIR ] ; then
```

Comment:
let's move the block to some function to keep the main function short and declarative.